test/sql/pragma/test_custom_optimizer_profiling.test

```diff
@@ -0,0 +1,106 @@
+# name: test/sql/pragma/test_custom_optimizer_profiling.test
+# description: Test PRAGMA custom_profiling_settings with optimizers
+# group: [pragma]
+
+require json
+
+statement ok
+PRAGMA enable_verification;
+
+statement ok
+PRAGMA enable_profiling = 'json';
+
+statement ok
+PRAGMA profiling_output = 'test/sql/pragma/output.json';
+
+# Turn on all optimizer settings
+statement ok
+PRAGMA custom_profiling_settings='{"ALL_OPTIMIZERS": "true"}'
+
+statement ok
+select unnest(['Maia', 'Thijs', 'Mark', 'Hannes', 'Tom', 'Max', 'Carlo', 'Sam', 'Tania']) as names order by random();
+
+statement ok
+PRAGMA disable_profiling;
+
+query I rowsort
+SELECT unnest(res) from (
```

Comment:
Let's add an `ORDER BY` to this query.

Reply:
oh I missed the `rowsort` - I am actually unsure what it does, I do not thing it sorts the column?
```cpp
	// perform any required query sorts
	if (sort_style == SortStyle::ROW_SORT) {
```

Reply:
In the [docs](https://duckdb.org/docs/dev/sqllogictest/result_verification#result-sorting).

But looks like `order by` is preferred, so I'll change it. Also in `test_custom_profiling_settings.test`

Reply:
sounds good 👍